WPMedia/fusion-news-theme-blocks
# TMEDIA-430 nav section re-render

## Description
Reduce re-renders of nav section children in header nav chain

## Jira Ticket
- [TMEDIA-340](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-430&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
Reduce render cycles of child features in the header nav chain

Reduce render cycles of the header nav chain main component where possible

## Test Steps

1. Checkout this branch `git checkout TMEDIA-430-rerender-header-nav`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/header-nav-chain-block`
3. Use fusion image  FUSION_RELEASE=3.1.0-use-content.0 based on work with https://arcpublishing.atlassian.net/browse/PEN-2102 (thanks @badmintonkid )
4. Create a new page with only a header nav chain 
5. Add a custom child within that chain. For example: 

<img width="404" alt="Screen Shot 2021-08-10 at 15 17 48" src="https://user-images.githubusercontent.com/5950956/128929975-33511e2a-3271-46a3-a6f4-4d7996ac5b67.png">
<img width="1278" alt="Screen Shot 2021-08-10 at 15 17 20" src="https://user-images.githubusercontent.com/5950956/128929976-67b2cd3d-513a-4246-818e-8f47fe999a4b.png">
6. Run the react profiler. Watch for that child's name in the react render cycle. It should only be rendered once https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en

## Effect Of Changes
### Before
- Children of header nav chain would be rendered three times 

### After
- Children only rendered once 
- Reduced one re-render by localizing the state of the logo visibility 
- Reduced another re-render courtesy of jason young/badmintonkid by looking at the useContent causing a hook to change unncessarily as the same content was fetched twice
- Nav logo still hides then displays as before 

## Dependencies or Side Effects

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
